plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerObjectCreateBase.java

```diff
@@ -222,6 +222,9 @@ protected IStatus run(DBRProgressMonitor monitor) {
                 return Status.CANCEL_STATUS;//GeneralUtils.makeErrorStatus("Null object returned");
             }
             monitor.beginTask("Save " + newObject.getClass().getSimpleName(), 3);
+            var closure = new Object() {
+                DBNDatabaseNode node = null;
+            };
```

Comment:
Why not the common pattern with a single-element array?
```java
DBNDatabaseNode[] node = {null};
```

Your approach will create a new inner anonymous class file.

Reply:
I don't like this pattern, it looks bad. Of course, if you insist, I can change it,

Reply:
Also that's what IDE suggest :)

Reply:
Any approach introducing a heap-allocated object with a reference will work. The array version is very common and doesn't introduce overhead like an anonymous class.

Reply:
OK